### DISCUSSION TITLE
Source code not found on UI since upgrade from 2.6 to 2.7

### DISCUSSION THREAD

**Discussion by William-Yi-Weng** [apache/airflow#34289] [Q&A]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow version 2.7
We get ``An error occurred while fetching the dag source code.`` on airflow web user interface, and it gets 404 Dag source not found from api call since we upgrade our existing airflow service from 2.6 to 2.7.
We have about 1600 DAGs sharing about 10 DAGs templates. I can see all DAGs running correctly and the DAGs details show well. And the dags folder is attached correctly in the web server container.

<img width="457" alt="image" src="https://github.com/apache/airflow/assets/46487209/d443f12a-fce9-40d3-bffc-4334425a9d84">

<img width="922" alt="image" src="https://github.com/apache/airflow/assets/46487209/a515725f-1b62-46c9-aa1b-9bed592be2dc">

### How to reproduce

It seems triggers exception of ``raise NotFound("Dag source not found")`` on this line.
https://github.com/apache/airflow/blob/38ecd4c4da65310f9461df769d274cb78b122630/airflow/api_connexion/endpoints/dag_source_endpoint.py#L40

Not sure it is caused by the secret key or the DAGs folder location (we are using secret_key from ssm parameter)

### Operating System

Debian GNU/Linux 12 - docker image

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==8.5.1
apache-airflow-providers-apache-hive==6.1.4
apache-airflow-providers-celery==3.3.2
apache-airflow-providers-common-sql==1.7.0
apache-airflow-providers-ftp==3.5.0
apache-airflow-providers-http==4.5.0
apache-airflow-providers-imap==3.3.0
apache-airflow-providers-postgres==5.6.0
apache-airflow-providers-redis==3.3.1
apache-airflow-providers-sqlite==3.4.3
apache-airflow-providers-ssh==3.7.1

### Deployment

Other Docker-based deployment

### Deployment details

We bundle the docker image from AZDO and deploy it into AWS ECS fargate task.
DB: RDS Postgresql-12.14
DAGs volume: EFS

### Anything else

I enabled debug log, but nothing found.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
> we are using temporary secret_key - empty value

Personally I have no idea how you could avoid Flack error in case of missing/empty_key

```console
RuntimeError: The session is unavailable because no secret key was set.  Set the secret_key on the application to something unique and secret.
```

So basically you need to set without it webserver might be broken.

>  And the dags folder is attached correctly in the web server container.

Webserver do not use DAG folder since Airflow 2.0. DAG code stored into the DB table `dag_code` and the webserver just read it from this table.

**Reply by William-Yi-Weng** [reply]
Sorry, I forgot we just moved secret key from airflow.cfg to webserver environment variable - SECRET_KEY

**Reply by Taragolis** [reply]
In that case it strange, because if you change secret then it should invalidate previous sessions.

Maybe this sounds as naive advice but did you try reset cache of your browser, or use another one, or in incognito mode?

If it is not help just check again your webserver logs it should write errors in to the log. In additional you could set [DEBUG severity](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#logging-level) for webserver container. Additional logs might help to identify what the issue in your case

**Reply by William-Yi-Weng** [reply]
It only shows the following messages on debug mode, when I trigger the failure
```
DEBUG - Function Arguments: ['file_token']
DEBUG - Getting data and status code
DEBUG - Pepared body and status code (404)
```

**Reply by William-Yi-Weng** [reply]
it returns 404 (file is not found) when I use REST api api/v1/dagSources/{file_token} to get the source code. Does it mean the encrypted path of the file (file_token) has been changed since upgrade to airflow 2.7

**Reply by S-W-K** [reply]
@William-Yi-Weng 
May I ask if you have found a solution? My team also encountered the same issue when upgrading from Airflow 2.0.1 to 2.7.1. We can access all attributes of the DAG except for 'dag_code'."

**Reply by William-Yi-Weng** [reply]
Unfortunately I haven't found a solution yet. I find any of file_token from my DAGs are invalid and return 404 to me. I was trying to set ssl_context to none, but it cannot work.

**Reply by doxenix** [reply]
@S-W-K @William-Yi-Weng Did you found any solution? The problem is in Ariflow 2.7.x and unfortunately it seems that Airflow 2.8.0 has also the same issue.

**Reply by doxenix** [reply]
In our case, the status code was **Error 403** - visible only in webrowser's console. 
After change of authorization policy to allow: [REST API PATH](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_dag_source), the issue gone.

**Reply by stanvv** [reply]
Same here (404 error), anyone found the root cause and solution?

**Reply by stanvv** [reply]
@doxenix @William-Yi-Weng we just solved this issue in our environment. The issue was that the AIRFLOW__WEBSERVER__SECRET_KEY had a different value than our flask secret key. Not sure why it did work in older versions though. FYI: we are using the community helm chart.

**Reply by potiuk** [reply]
We closed a few security issues related to it in recent versions of Airflow one of them this one - so it looks like you simply had "accidental" access to the source code even if your configuration was wrong: https://github.com/apache/airflow/pull/36257

**Reply by S-W-K** [reply]
> @doxenix @William-Yi-Weng we just solved this issue in our environment. The issue was that the AIRFLOW__WEBSERVER__SECRET_KEY had a different value than our flask secret key. Not sure why it did work in older versions though. FYI: we are using the community helm chart.

Same solution!  We found that AIRFLOW__WEBSERVER__SECRET_KEY has different values in several manifests in our environment . We changed them to the same value then the Error has gone.

**Reply by fnsalinas** [reply]
> AIRFLOW__WEBSERVER__SECRET_KEY

@S-W-K @stanvv 

We're facing the same issue, please help us leting us know ¿how did you found the right AIRFLOW__WEBSERVER__SECRET_KEY?

Thank you in advance!

**Reply by stanvv** [reply]
> > AIRFLOW__WEBSERVER__SECRET_KEY
> 
> @S-W-K @stanvv
> 
> We're facing the same issue, please help us leting us know ¿how did you found the right AIRFLOW__WEBSERVER__SECRET_KEY?
> 
> Thank you in advance!

I'm not sure what your setup is, but ours is AKS, so just login to the web pod (`kubectl exec ... `) and `echo $FLASK_SECRET_KEY` and `AIRFLOW__WEBSERVER__SECRET_KEY`. There we saw they were different.

Eventually we just added this part to make them equal:
```
  extraEnv:
    - name: AIRFLOW__WEBSERVER__SECRET_KEY
      valueFrom:
        secretKeyRef:
          name: airflow-core-secrets
          key: FLASK_SECRET_KEY    
    - name: FLASK_SECRET_KEY
      valueFrom:
        secretKeyRef:
          name: airflow-core-secrets
          key: FLASK_SECRET_KEY    
```
